internal/config/config_type.go

```diff
@@ -271,7 +271,34 @@ func (c *fileConfig) GetWithSource(hostname, key string) (string, string, error)
 	return value, defaultSource, nil
 }
 
+func validConfigEntries(key string) []string {
+	configEntries := make(map[string][]string)
+
+	configEntries["git_protocol"] = []string{"ssh", "https"}
+
+	return configEntries[key]
+}
+
+func validateConfigEntry(key, value string) error {
+	validValues := validConfigEntries(key)
+
+	if len(validValues) == 0 {
+		return nil
+	}
+
+	for _, v := range validValues {
+		if v == value {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("invalid value for %s", key)
```

Comment:
Could the error message also list possible values so it's more helpful to the user?

Reply:
@mislav Will this work?

```go
return ​fmt​.​Errorf​(​"%s is not a valid value for %s, supported values: %v"​, ​value​, ​key​, ​validValues​)
}
```